src/Processors/Executors/PipelineExecutor.cpp

```diff
@@ -364,13 +364,12 @@ void PipelineExecutor::initializeExecution(size_t num_threads, bool concurrency_
         /// To avoid counting them in ConcurrencyControl, we create dummy slot allocation.
         cpu_slots = grantSlots(num_threads);
     }
-    size_t use_threads = cpu_slots->grantedCount();
 
     Queue queue;
     Queue async_queue;
     graph->initializeExecution(queue, async_queue);
 
-    tasks.init(num_threads, use_threads, profile_processors, trace_processors, read_progress_callback.get());
+    tasks.init(num_threads, 1, profile_processors, trace_processors, read_progress_callback.get());
```

Comment:
Do you mean making the threashold adjustable by parameter?

Reply:
no, i propose to start from `TOO_MANY_IDLE_THRESHOLD` instead of 1

Reply:
`use_threads` is used to determine the tasks finished condition (whether all use_threads are into idle). It should reflect the number of threads spawned.  Make it to `TOO_MANY_IDLE_THRESHOLD` do not affect the performance. Furthermore, it might greater than `num_threads` (e.g `number_threads = 2`), which will make tasks finished condition unmet forever.